### DISCUSSION TITLE
Interoperability with Mastodon/ActivityPub

### DISCUSSION THREAD

**Discussion by eloquence** [bluesky-social/atproto#1716] [General Discussion]
What are the current plans from BlueSky (the company) for interoperability with Mastodon/ActivityPub?

We currently have two emerging alternatives to X that deserve to be taken seriously: the fediverse and BlueSky. The fediverse tends to attract folks who highly value open source in principle and practice; BlueSky tends to attract folks who are looking for a drop-in Twitter replacement.

In this context, I would argue that a clear commitment from the key parties (that includes @gargron's Mastodon gGmbH and Bluesky the company) towards an interoperable social web is essential. We should be able to follow each other no matter where we decide to make our social media home. Even Mark Zuckerberg's Threads has at least made a commitment towards interoperability.

Is Bluesky (the company) willing to make such a commitment or has it already done so in venues where I've not seen it? The last comment from anyone associated with the company I've been able to find is this from @pfrazee [here](https://github.com/bluesky-social/atproto/issues/345#issuecomment-1751302322) in November 2022:

>  currently, no there's no plan for AP integration in the Bluesky client.

Is that still all that can be said? Even a clear commitment that has not yet been translated into the roadmap would be highly valuable at this stage.

**Comment by snarfed** [comment]
Hi Erik! Important question. It's been discussed a ton, both elsewhere and here, eg https://atproto.com/guides/faq#why-not-use-activitypub, https://github.com/bluesky-social/atproto/issues/255, https://github.com/bluesky-social/atproto/issues/345, https://github.com/bluesky-social/atproto/discussions/1656, etc, but always good to keep talking!

At the risk of putting words into the team's mouths, their position so far has generally been:
1. ActivityPub is great! Decentralized social networks in general are great. And especially now, given all the upheaval in the space, this is a great opportunity to try a bunch of different things and see what works, before we try to consolidate too much.
1. ATProto specifically is trying to do a few big things differently, notably account portability, "big world" services, and composable moderation and algorithms, all of which are difficult or impossible in ActivityPub or conflict with the fediverse's current culture and norms. (This is basically straight from the FAQ link above.)
1. Building ATProto and Bluesky alone is a ton of work. Bridging and compatibility may be possible eventually, but aren't short or medium term priorities.

I largely agree with that stance. Fortunately, the team is supportive of external efforts to bridge the protocols, which is what I've been working on with [Bridgy Fed](https://fed.brid.gy/). [Recent status update here](https://snarfed.org/2023-10-06_bridgy-fed-status-update-7); [some technical details here](https://fed.brid.gy/docs#compare). Feedback and contributions are welcome!

**Reply by ElanHasson** [reply]
What is a "big world" service?

**Reply by snarfed** [reply]
Sorry, that's a term that the Bluesky team used to use to describe parts of the ATProto architecture, and now maybe uses less often. It's any part that aggregates a global or near-global view of the network in some way, eg relays, appviews, search indexes, etc.

**Comment by eloquence** [comment]
Thanks - I completely understand the practical difficulty of getting to millions of users and making the protocol work, and having to prioritize that over interoperability work.

However, I do think it's crucial to make certain interoperability commitments early, so they can be factored into design decisions. Some things will probably never work across servers, and that's fine. For example, maybe certain kinds of posts just can't be federated (or might need to be shortened or replaced by placeholder).

An example approach for that could be to identify some key user stories, e.g.
- following users and seeing text posts
- following users and seeing image/video posts
- reporting a user
- blocking/muting a user

and stating clearly which ones are in scope for long term interoperability goals and which ones aren't. Further, even knowing that interoperability (with qualifiers) _is_ a long term goal for the project, and what conditions will need to be satisfied before it can be prioritized by Bluesky the company, would be useful.

None of this is to take away from your bridging efforts, which I agree are important and useful. However, there is a risk of bridges falling down, especially when there is no first-party commitment to interoperability from the biggest node operators & companies supporting the respective projects. Beyond that, bridges can add further UX hurdles to an already complicated idea (following users on another server).

**Reply by snarfed** [reply]
Thanks for the kind words! And absolutely agreed, bridging is always awkward, worse UX, and a degraded experience compared to native compatibility.

I'll mostly defer to the team on whether they want to discuss/answer this again. Historically though, so far they've generally said that native AP/fediverse compatibility is not on their roadmap. Beyond the reasons above, one way to think about it is that the protocols are so different that you can't just add compatibility. They'd have to build a full fledged bridge themselves, internally, to be compatible. And that internal bridge would be awkward in many of the ways that my external bridge will be - identifiers and handles don't translate intuitively, composable moderation labels and feed algorithms may not translate at all, fediverse users would hate the idea of their profiles and posts being ingested by Bluesky BGSes/AppViews, etc.

Also, effort wise, we can't just estimate story points for a handful of user stories, pick the top few, and optimize ROI that way. Due to AT Protocol's design itself, going from 0 to 1 to translate _anything_ external into ATProto takes a ton of work. It's like if the initial mandatory translation layer was scoped at 100 story points, _and then_ following was 5 story points, posts were 7, reporting was 2, etc.

I'm obviously with you on interop! Bridge all the things. Just trying to explain what their position has been (and probably still is), and why it makes sense to me at least.

**Comment by bnewbold** [comment]
There are a bunch of ideas, possibilities, and open questions here, which make it difficult to give a direct answer.

Can ActivityPub/Mastodon content be transformed and represented in atproto, and vica versa? Seems pretty likely! atproto has a flexible schema system ("Lexicons") and it should be possible to make a pretty direct representation of the relevant schemas used by Mastodon. Likewise ActivityPub allows new event schemas, and the `app.bsky.*` schemas could be represented there. The "post" content is also fairly similar and could be translated without perfect schema mapping. It should not be hard to do things like export posts and re-import in the other system... if issues like references, threading, and social graph are ignored.

Can subscriptions flow between the two protocols, at scale? This seems harder. ActivityPub has a concept of per-server inboxes and outboxes between individual accounts. atproto has a concept of aggregating global public content in to a unified firehose, and indexing that. Those are pretty different network architectures to reconcile.

Can core expectations of privacy, safety, and moderation be respected across both protocols? This seems like a challenge. atproto is currently entirely globally public and enumerable, including blocks. ActivityPub/Mastodon has a lot more nuance: a lot of information is openly accessible, or can be scraped or inferred, but there is meaningful friction involved. Naive following and feeds is very simple: unauthenticated RSS has been implemented for atproto (at least as a demo), and I assume the same for ActivityPub/Mastodon. But how do blocks, mutes, labeling, blocklists, reply controls, and future moderation/interaction rules work out? These are somewhat challenging to ensure even within the same protocol.

Can clients support both? Yes, there are already multiple clients which can fan out a post to multiple networks, and browse content from multiple networks.

Can folks reply across protocol? Eg, a hybrid reply thread? This seems hard without some form of bridging.

Can graph aggregations be computed across protocols? Eg, like and follow account. Probably, though doesn't seems like a super high priority.

Can identity be mapped across both protocols? Superficially, "in many cases", sure. But the underlying properties are different, and things would break down with account migration, which are an important feature of both protocols.

Will there be some way for Bluesky or any atproto service to implement direct compatibility with ActivityPub standard, or vica-versa? In a meaningful way? Seems like a stretch. As a direct answer, phrased this way, it is not on the Bluesky roadmap.

All of the Fediverse (the community/ecosystem), Mastodon (the company and project), and Bluesky (the company) are committed to interoperability with their respective protocols; to giving users control over their own experiences by using user agents (clients) of their own choosing; and to giving users access to their own content. Those are the foundations for any kind of bridging or cross-protocol interactivity, and at least from my perspective they are solidly in place.

I'm personally pretty excited about multi-protocol clients and simple syndication bridges (eg, for public broadcast usecases). The work @snarfed and others are doing on bridging is great. I think that working out identity, privacy, and safety invariants across protocols could be possible, but would be challenging.

**Reply by qazmlp** [reply]
This is a very late reply, but I just want to comment on this particular bit:

> Naive following and feeds is very simple: unauthenticated RSS has been implemented for atproto (at least as a demo), and I assume the same for ActivityPub/Mastodon.

That is in fact not very close to universally feasible already, because all follows are authenticated and also (potentially manually) authorised on ActivityPub.

A significant number of instances do publish RSS feeds for a subset of an account's public activity, but I *think* this is far from universal and likely depends on settings. It also wouldn't be efficient to use polling to detect updates on ActivityPub, since you'd have to re-ingest everything to detect deletion of older posts in many cases. (There is a public ActivityPub outbox on some services but not all, and I'm unsure if it lists post deletions or could be reliably used to detect account deletion.)

Bridgy Fed gets around this by using a bot account to follow the bridged account, but this is notably done only *after* the account in question agrees to bridging.

Flipboard (.com) recently tried to ingest posts proactively, i.e. by mass-following many accounts from a (nondescript) service account in order to allow their users to follow ActivityPub accounts known to the flipboard.social instance without an own presence on ActivityPub. This was seen as a violation of the social contract and led to widespread defederation of flipboard.com (and frequently also defederation or limiting of flipboard.social).
They did apologise, but it could take years for their platform to regain trust and wide connectivity on the ActivityPub fediverse.

**Edit**: It's Flipboard, sorry. I keep misremembering the name.

**Comment by eloquence** [comment]
Thank you both for the thoughtful responses, @snarfed and @bnewbold, I appreciate it.

Broadly speaking, my understanding is that both Mastodon and BlueSky are positioning themselves as public benefit projects -- Mastodon explicitly as a nonprofit, BlueSky with leadership by a PBLLC. I hope this public benefit orientation will lead us to a social web that strikes a good balance for users and isn't a zero-sum game.

Beyond Mastodon, the ask for interoperability is likely to get louder if Threads follows through on its promise of implementing ActivityPub. I think it would be helpful to users of all three platforms if interoperability goals could be made as explicit as possible, even if they are very narrowly scoped to begin with.

> simple syndication bridges (eg, for public broadcast usecases)

A project-level commitment to support read-only consumption of updates from ActivityPub accounts (without needing two logins) would be a great start upon which to iterate, IMO. This would already help for many use cases, such as public sector accounts for emergency services, as well as news site accounts. 

> I think that working out identity, privacy, and safety invariants across protocols could be possible, but would be challenging.

No question. I hope down the road, the idea of a cross-project task force to examine such questions may become viable. In the meantime, even modest gains will help the users of both platforms, and signal that we're building the social web together and not apart.

**Reply by SlashScreen** [reply]
Assuming Threads becomes a major player in the social media game. I personally don't think so, because everyone I know of who made a Threads account no longer uses it.
I personally think Tumblr's proposed ActivityPub interop would be a larger force, but I also call into question if they will ever go through with it (especially since their backend team is lacking from what I understand)

Nonetheless, I get your point, and none of what I said actually matters in this case.

**Comment by erlend-sh** [comment]
A simple interop MVP I don’t see mentioned a lot is the ability to fully embed an apub/atproto post on its opposite platform. This would allow for a kind of quote-posting across networks.

Like if I posted this link in a Bluesky post: https://mastodon.social/@pfefferle/111217236362579201

..then it should be possible to display that as an [oEmbed](https://docs.joinmastodon.org/methods/oembed/).

If that could somehow be combined with a [WebMention](https://www.w3.org/TR/webmention/) to inform me that my apub post has been shared elsewhere (and vice versa), we’d have significantly greater cross-border connectivity than the status quo.

**Reply by bnewbold** [reply]
That is a concrete suggestion that seems doable. The WebMentions could be generated for the entire atproto network by a service that consumes the global firehose (less coordination/adoption, more centralized), or by PDS or AppView implementations.

Going the other way (ActivityPub mention of an atproto record results in atproto back-notification) might be harder. Atproto requires a record in a repo for things like mentions and backlinks; it isn't clear which or where such a mention record should be created for an inbound WebMention. Perhaps the PDS could create an inbound mention record in the "target" repo itself?

**Reply by snarfed** [reply]
Nice! Fwiw https://fed.brid.gy/ is a multi-protocol bridge, and already supports webmention + microformats2, the latter being just as important as the former. You could send a webmention with a source page on a public bluesky client, but if that client doesn't server-side render HTML that includes microformats2, webmention receivers won't be able to do much with the source other than say, "hey, this page linked to us."

Native is always nicer than bridge when possible though! Happy for BF to serve as a proof of concept and answer any q's when the time comes.

**Reply by SlashScreen** [reply]
I like this idea as a first step.

**Reply by erlend-sh** [reply]
Since someone bumped the thread, I’ll go ahead and nudge on this:

> *from @bnewbold :*
> 
> Going the other way (ActivityPub mention of an atproto record results in atproto back-notification) might be harder. Atproto requires a record in a repo for things like mentions and backlinks; it isn't clear which or where such a mention record should be created for an inbound WebMention. Perhaps the PDS could create an inbound mention record in the "target" repo itself?

With a year passed, might there be some new avenues to explore here now?

**Comment by upintheairsheep** [comment]
As a ActivityPub user, I’d very much want to have support to like, reply, get replied to, and see BlueSky posts, while I don’t care if they don’t see non-reply ActivityPub posts. Failure to bridge it with the greater ActivityPub will lead to a “digital apartheid” of sorts, just like how social media is today closed down.

**Reply by SlashScreen** [reply]
not sure what you mean by Apartheid here

**Comment by upintheairsheep** [comment]
A “digital wall” planted that separates primarily fediverse users from
being able to read and communicate with atproto users.

**Reply by Espionage724** [reply]
I agree. I've heard of Bluesky recently, decided to _finally_ check it out in self-hosting style, but I'm not about to get into a one-guy protocol system. I just assumed Bluesky was using ActivityPub since everyone else in this niche ordeal does.

Nah this is straight-up centralized, and I'll likely continue hosting Friendica with ActivityPub support.

**Comment by dizorg** [comment]


**Reply by dizorg** [reply]
Or pubsubhubbub ?

**Comment by pixiekat** [comment]
I'm a big fan of both Bluesky and Mastodon/Fediverse so I would definitely like to see some sort of integration, if only to keep from having to segregate my followers between two different platforms. I think the implosion of Twitter kind of showed us how fragmentation of social networks is bad. Even if it's not full integration, some sort of ability to read and share posts between the two would be great.

**Reply by surfdude29** [reply]
[This post on the ATProtocol Dev Blog](https://atproto.com/blog/feature-bridgyfed) interviewing the author of Bridgy Fed (@snarfed who commented above) says:

> It's fully bidirectional; from any supported network, you can follow anyone on any other network, see their posts, reply or like or repost them, and those interactions flow across to their network and vice versa. [More details here.](https://snarfed.org/2023-11-27_re-introducing-bridgy-fed)

**Comment by swarming1905** [comment]
I'm not a dev or programmer, just dropping by to say that I'm not fussed in the short-medium term, but in the long run, given the scale at which ActivityPub is already being used (such as on Lemmy/Kbin, PixelFed, now Flipboard, even Tumblr seems to be working towards it), my worry is that without gradual process towards interoperability we could just end up with two mutually exclusive protocols, and also that ATProto/Bluesky could get left behind due to mass early adoption of ActivityPub.

**Comment by mount2010** [comment]
Hi. I'm a nobody to this project, but I'm concerned about the future of the internet, which I hope open, democratic platforms like these will improve. Currently, I am a user of ActivityPub platforms, and I would prefer not to join Bluesky or an AT-powered platform at this moment. I find interoperability between these platforms important, and almost critical, as I see Bluesky/AT increase in popularity. The internet does not need more segmentation, and the Fediverse was meant to give us the good parts of segmentation without the bad parts. AT and AP do not and should not need to compete for users. 

As a user of ActivityPub platforms, I would very much enjoy being able to interact with users on Bluesky and other AT powered platforms. I urge the developers of the AT protocol to consider interoperability, at least at the basic level, more important.

**Comment by upintheairsheep** [comment]
That’s what I’m saying. No AP integration will just fragment us.

**Comment by MonikaBarget** [comment]
I think this is a fascinating and important issue. I, personally, would be pleased with read-only access to BlueSky content. Since the decline of Twitter / X, several academic accounts that I follow to find out about events, calls for papers and new publications have moved to BlueSky while I am a Mastodon user by conviction. I do not necessarily need to respond to the accounts on BlueSky -- I just don't want to miss out on their posts. I think this is pretty representative of academic social media usage more generally.

But aside from personal experience, I would like to share the following article on the SkyBridge tool that allows BlueSky users access to some Mastodon apps but, according to the author, also permits "accessing Bluesky feeds via the Mastodon lists screen": https://joeross.me/2023/06/26/skybridge-is-a.html

Have any of you got more insights?

**Reply by goliv04053** [reply]
Not related but kinda related:

https://snarfed.org/2024-02-12_52106 (A guy developed a bridge between Fediverse and Bluesky)

Website related too:
https://fed.brid.gy/

**Reply by bnewbold** [reply]
If you are looking just to follow top-level posts (not replies) from bsky accounts, and have them show up in mastodon, you should be able to cobble this together today using the public bsky RSS feeds (per account; RSS URL is in the HTML metadata on public bsky.app account profile views) and activitypub tools like RSS Parrot.

**Reply by TahomaSoft** [reply]
I'm working on transmogrifying fediverse posts to Bluesky using the RSS feed function (in Mastodon, and probably other Fediverse servers).  Rather relevant for this conversation is an observation that ActivityPub, Mastodon, and other fediverse server software can do more on their side to promote interoperability.  Not only to ATproto, but also within ActivityPub. 

For example, take content warnings. The RSS feed structure doesn't have a field for content warnings; it is scrambled into the main text post. To grok it, the html text field of the RSS feed has to be parsed specially. Which is a bit of a pain and a back-[]ssward 'feature'.

**Comment by goliv04053** [comment]
Commented on the other comment before mine but i'm going to make other post to be more visible.

https://snarfed.org/2024-02-12_52106 (A guy developed a bridge between Fediverse and Bluesky)

Website related too:
https://fed.brid.gy/

**Reply by DavidBuchanan314** [reply]
no need to make it more visible, he's already the first reply in this discussion ;)

**Comment by snarfed** [comment]
Idle thought: for everyone here interested in AT Protocol <=> ActivityPub interop, a fun weekend project would be to write a set of lexicons for ActivityStreams 2 activities and objects!

- https://atproto.com/specs/lexicon
- https://www.w3.org/TR/activitystreams-core/
- https://www.w3.org/TR/activitystreams-vocabulary/

**Reply by bnewbold** [reply]
Yup! See also: https://berjon.com/ap-at/

And DSNP uses some subset of ActivityStreams as part of their protocol, might be worth checking out how they did things: https://spec.dsnp.org/

**Reply by outlaw-dame** [reply]
I wonder has this been done yet?

**Reply by snarfed** [reply]
@mikekasprzak did this in https://github.com/bluesky-social/atproto/pull/3943 ! So cool!

https://github.com/mikekasprzak/atproto/tree/2cf1506ac394b0ed5680b094ccfa5ff090f29ff7/lexicons/org/w3/activitystreams

https://bsky.app/profile/distraction.engineer/post/3lr2qwla2rs2w

**Comment by RickyRomero** [comment]
Threads has begun [rolling out ActivityPub support to more users](https://www.theverge.com/2024/3/21/24107881/threads-fediverse-beta-launch-mastodon). Bluesky users would benefit from being able to follow people on Threads and vice-versa.

**Reply by pixiekat** [reply]
Yes! Would love to see it. It would only help both Bsky and Threads grow now that Twitter is dead.

**Reply by hudsonm62** [reply]
sadly i don't see this ever happening (officially)

**Comment by erlend-sh** [comment]
Making some progress on overlapping identity tech:
- https://writing.exchange/@erlend/112121718112283325
- https://were.social/objects/14436e95-6740-4fd8-811f-7453ba81ad5a
- https://were.social/objects/669c1495-387b-41b8-89cc-bb2a4f32f6af
- https://social.coop/@bnewbold/112146650520103588

[Cross-posted](https://socialhub.activitypub.rocks/t/nomadic-identity-for-the-fediverse/2101/65?u=erlend_sh).

**Comment by goliv04053** [comment]
Any updates?

**Reply by andypiper** [reply]
From what perspective?

**Comment by MonikaBarget** [comment]
In the meantime, I have tested the [RSS Parrot](https://mastodon.social/@birb@rss-parrot.net) to follow a BlueSky account on Mastodon, but it did not work. I asked what I did wrong but didn't get a helpful answer. The owner of my Mastodon server wrote:

"Can't speak to any third-party integrations, sorry. [@akademienl](https://mastodon.social/@akademienl@akademienl.social) is running smoothly afaik - any integrations would be client-side or on someone else's server."

I cannot even access the RSS Parrot website given in their Mastodon account because of a 502 error: https://rss-parrot.net/

If anyone knows of better solutions to at least one-sidedly follow BlueSky accounts, I would appreciate your advice. Most of my academic colleagues have migrated to BlueSky, but I don't like the service and would - also for political reasons - love to keep using Mastodon, but it feels lonely at the moment.

**Reply by JustinLex** [reply]
See the above mentions of Bridgy Fed

**Comment by TahomaSoft** [comment]
I created a cross-poster using RSS to go the other way, from Mastodon to BlueSky.It's here:https://github.com/TahomaSoft/SaltstraumenSent from my iPhoneOn Sep 3, 2024, at 07:32, Monika Barget ***@***.***> wrote:﻿
In the meantime, I have tested the RSS Parrot to follow a BlueSky account on Mastodon, but it did not work. I asked what I did wrong but didn't get a helpful answer. The owner of my Mastodon server wrote:
"Can't speak to any third-party integrations, sorry. @akademienl is running smoothly afaik - any integrations would be client-side or on someone else's server."
I cannot even access the RSS Parrot website given in their Mastodon account because of a 502 error: https://rss-parrot.net/
If anyone knows of better solutions to at least one-sidedly follow BlueSky accounts, I would appreciate your advice. Most of my academic colleagues have migrated to BlueSky, but I don't like the service and would - also for political reasons - love to keep using Mastodon, but it feels lonely at the moment.

**Reply by MonikaBarget** [reply]
Your link did not work but I found the repo here: https://github.com/TahomaSoft/Saltstraumen

Unfortunately, the Readme only has the title, and I cannot find the documentation. Above all, what should I tell people on BlueSky who want to follow me? What do they have to look for to see my content?

**Comment by TahomaSoft** [comment]
It's intended to run as a process somewhere (a linux machine at home, for example), taking your posts on Mastodon and cross posting to your account on bluesky. It is written in python, and is pretty rough around the edges.ErikSent from my iPhoneOn Sep 3, 2024, at 07:46, Monika Barget ***@***.***> wrote:﻿
Your link did not work but I found the repo here: https://github.com/TahomaSoft/Saltstraumen
Unfortunately, the Readme only has the title, and I cannot find the documentation. Above all, what should I tell people on BlueSky who want to follow me? What do they have to look for to see my content?

**Comment by nickpassaro** [comment]
I know this is an old discussion, but for those unaware, Meta has recently changed their moderation policies to be in line with X (Twitter) which is to say that they will no longer be moderating anything other than illegal content. This matters to this discussion because as more and more users will inevitably begin fleeing Threads and Instagram in the coming months. As Meta platforms end up becoming just as toxic as Twitter, I think interoperability between Bluesky and ActivityPub in a way that is built into the Bluesky experience should be more seriously considered. Even if the interoperability directly uses Bridgy Fed and only offers its features, it would be nice if Bluesky users could opt into and opt out of ActivityPub discoverability at will. I don't know if this is more of a suggestion for Bluesky itself and not for the ATProto, but for smaller communities where people don't want to have to join multiple platforms to be able to keep up with each other, this kind of interoperability would be really beneficial, even if it's very basic. As more people start fleeing Meta in the coming months, as they've already left Twitter, communities will find themselves fragmenting and this could help alleviate that.

**Reply by TahomaSoft** [reply]
Thank you for your insight and suggestions!

I've got some ideas on how to revamp my code to be more useful.

**Reply by snarfed** [reply]
> Even if the interoperability directly uses Bridgy Fed and only offers its features, it would be nice if Bluesky users could opt into and opt out of ActivityPub discoverability at will.

FWIW, Bridgy Fed already lets Bluesky users opt in and out of it, and AP discoverability, at will!

**Reply by outlaw-dame** [reply]
Snarfed mentioned incompatible cultures many times and I’ve seen people just gloss over it, I’m at a loss as to why?

**Comment by Goldmaster** [comment]
Where is Bluesky up to with activity pub support?

**Reply by snarfed** [reply]
Short answer is, they're not. See discussion above.

**Reply by snarfed** [reply]
Feel free to try https://fed.brid.gy/ !

**Reply by Goldmaster** [reply]
I assume you mean this post https://github.com/bluesky-social/atproto/discussions/1716#discussioncomment-7213000

```3. Building ATProto and Bluesky alone is a ton of work. Bridging and compatibility may be possible eventually, but aren't short or medium term priorities.```

The problem with bridgy fed is that the other user has to follow bridgy fed for posts to show up. Trying to explain or convince other social media managers to do something as simple as follow a profile to improve engagement is like explaining how pressing the accelerator to drive forward works. Very often, social media managers want to improve the follower to following ratio. Bluesky could follow bridgy fed by default when new accounts join or admins could set that every account follows bridgy fed.

I guess it could happen but not next year sorta thing.

**Comment by msonnb** [comment]
I built [Fedisky](https://github.com/msonnb/fedisky), a PDS sidecar service that you can install alongside your self-hosted PDS. It exposes users as ActivityPub actors (@alice.fedisky.social on Bluesky becomes @alice@fedisky.social on Mastodon), stores incoming follows and then delivers posts as ActivityPub notes to your fediverse followers. It also bridges Fediverse replies back to Bluesky using a service account (@mastodon.fedisky.social for example).

It's still under development but you can try it out using the [install script](https://github.com/msonnb/fedisky/blob/main/fedisky-installer.sh) and leave feedback if you want 🙏 . Here's a quick demo: https://www.youtube.com/watch?v=u-U0jMMXRkE

**Reply by snarfed** [reply]
Whoa, this is awesome! Congrats on launching it! Do you have an example instance running it that we can see?

**Reply by snarfed** [reply]
Also, check out https://blog.anew.social/bridging-identity-with-account-links/ if you haven't already, it's how we're trying to make all of the different bridges interoperate. We'd love to have you use it too!

**Reply by msonnb** [reply]
> Whoa, this is awesome! Congrats on launching it! Do you have an example instance running it that we can see?

Thanks! I have it deployed on fedisky.social with some test posts on the `alice` account (see [Bluesky](https://bsky.app/profile/alice.fedisky.social) and [Mastodon](https://mastodon.social/@alice@fedisky.social)). Registrations are currently closed though.

> Also, check out https://blog.anew.social/bridging-identity-with-account-links/ if you haven't already, it's how we're trying to make all of the different bridges interoperate. We'd love to have you use it too!

Nice! I just added the ATProto handle to the actor document. Other way around (actor id in DID document) looks a bit more tricky due to the PLC operation and email verification required by the user, going to have to figure out how this could work from a UX perspective.

**Reply by snarfed** [reply]
> Other way around (actor id in DID document) looks a bit more tricky due to the PLC operation and email verification required by the user

Yeah, either that or modify the PDS itself. Not a big deal, even just having the DID in the AP actor's `alsoKnownAs` is a great step!

**Reply by quillmatiq** [reply]
This looks awesome, @msonnb! Excited to see where you take this. Looks like @snarfed has you covered on interop with Bridgy Fed and other bridges, but whenever you're ready, it would be neat to figure out whether we can pull something like this into [Bounce](https://bounce.anew.social/) as well, even if it means a separate version for FediSky.

**Reply by bmann** [reply]
Looks cool! If you want something that isn't purely a dev discussion, come introduce yourself in the community forum https://discourse.atprotocol.community
